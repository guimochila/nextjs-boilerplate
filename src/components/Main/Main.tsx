import * as S from './styles'

const Main = ({
  title = 'React Pro',
  description = 'Typescript, ReactJS, NextJS and Emotion',
}) => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.svg"
      alt="Image of an atom and Advanced React written on its right side"
      width={250}
      height={55}
    />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="Developer with a screen in front of him"
      height={260}
      width={300}
    />
  </S.Wrapper>
)

export default Main
