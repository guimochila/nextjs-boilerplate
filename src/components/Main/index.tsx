import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.svg"
      alt="Image of an atom and Advanced React written on its right side"
    />
    <S.Title>React Pro</S.Title>
    <S.Description>Typescript, ReactJS, NextJS and Emotion</S.Description>
    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="Developer with a screen in front of him"
    />
  </S.Wrapper>
)

export default Main
