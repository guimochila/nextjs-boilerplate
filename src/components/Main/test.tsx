import { render, screen } from '@testing-library/react'
import { matchers } from '@emotion/jest'

expect.extend(matchers)

import Main from './index'

describe('<Main />', () => {
  it('should render heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react pro/i }),
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyleRule('background-color', '#06092b')
  })
})
