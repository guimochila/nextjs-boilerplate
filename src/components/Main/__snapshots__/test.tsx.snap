// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render heading 1`] = `
.emotion-0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.emotion-2 {
  font-size: 2.5rem;
}

.emotion-3 {
  font-size: 2rem;
  font-weight: 400;
}

.emotion-4 {
  margin-top: 3rem;
  width: min(30rem, 100%);
}

<main
  class="emotion-0"
>
  <img
    alt="Image of an atom and Advanced React written on its right side"
    class="emotion-1"
    src="/img/logo.svg"
  />
  <h1
    class="emotion-2"
  >
    React Pro
  </h1>
  <h2
    class="emotion-3"
  >
    Typescript, ReactJS, NextJS and Emotion
  </h2>
  <img
    alt="Developer with a screen in front of him"
    class="emotion-4"
    src="/img/hero-illustration.svg"
  />
</main>
`;
